k9mail/src/androidTest/java/com/fsck/k9/activity/MessageReferenceTest.java

```diff
@@ -124,6 +127,11 @@ public void equalsWithMessageReferenceContainingDifferentUidShouldReturnFalse()
         assertEqualsReturnsFalseSymmetrically(messageReferenceOne, messageReferenceTwo);
     }
 
+    @Test
+    public void hashcodeVerifyEqualsContract() {
+        EqualsVerifier.forClass(MessageReference.class).suppress(Warning.NONFINAL_FIELDS).verify();
```

Comment:
As a general rule: If you're unsure whether or not to suppress a warning, don't!

You discovered the principal problem of mutable objects. The easy fix is to use [immutable objects](http://en.wikipedia.org/wiki/Immutable_object) whenever possible. `MessageReference` looks like it could be easily made immutable. So you should definitely do that instead of suppressing the warning.